parse/getast.go

```diff
@@ -339,14 +340,34 @@ func (fs *FileSet) getField(f *ast.Field) []gen.StructField {
 	if f.Tag != nil {
 		body := reflect.StructTag(strings.Trim(f.Tag.Value, "`")).Get("msg")
 		tags := strings.Split(body, ",")
-		if len(tags) == 2 && tags[1] == "extension" {
-			extension = true
-		}
+
 		// ignore "-" fields
 		if tags[0] == "-" {
 			return nil
 		}
-		sf[0].FieldTag = tags[0]
+
+		if tags[0] != "" {
+			sf[0].FieldTag = tags[0]
+		}
+
+		if len(tags) > 1 {
+			last := len(tags) - 1
+			extension = tags[last] == "extension"
+
+			var err error
+			switch tags[1] {
+			case "uint":
+				str, base := numeric(tags[0])
```

Comment:
`strconv.ParseInt(tags[0], 0, 64)` will infer hex or octal based on the prefix; no need to do that yourself.